redbot/cogs/mod/events.py

```diff
@@ -19,20 +20,30 @@ class Events(MixinMeta):
 
     async def check_duplicates(self, message):
         guild = message.guild
-        author = message.author
+        author = message.author.id
+        channel = message.channel
 
-        guild_cache = self.cache.get(guild.id, None)
-        if guild_cache is None:
-            repeats = await self.config.guild(guild).delete_repeats()
+        repeats = await self.config.guild(guild).delete_repeats()
+        repeats_channels = await self.config.guild(guild).delete_repeats_channels.all()
+        if str(channel.id) in repeats_channels:
+            if repeats_channels[str(channel.id)] == -1:
+                return False
+        else:
             if repeats == -1:
                 return False
-            guild_cache = self.cache[guild.id] = defaultdict(lambda: deque(maxlen=repeats))
 
+        guild_cache = self.cache.get(guild.id, None)
+        if guild_cache is None:
+
+            # Cache is Map (channel -> Map (author -> messages))
+            guild_cache = self.cache[guild.id] = await create_new_cache(self.config, guild)
         if not message.content:
             return False
 
-        guild_cache[author].append(message.content)
-        msgs = guild_cache[author]
+        guild_cache[str(channel.id)][author].append(message.content)
+        msgs = guild_cache[str(channel.id)][author]
+        print(guild_cache)
+        print("-----------------------------------------------------------------------")
```

Comment:
You should not be leaving **any** `print` statements in the code. If there is something that should be displayed in console, you can use the `logging` module to do so like other parts of the code. This is not such a case however, so these lines should be removed.